android/libraries/rib-android/src/main/kotlin/com/uber/rib/core/lifecycle/ActivityLifecycleEvent.kt

```diff
@@ -17,7 +17,34 @@ package com.uber.rib.core.lifecycle
 
 import android.os.Bundle
 
-/** Lifecycle events that can be emitted by Activities. */
+/**
+ * Lifecycle events that can be emitted by Activities.
+ *
+ * ### Ordering semantics
+ *
+ * This class implements [Comparable], but it does **not** override `equals`, so even though the
+ * following holds:
+ * ```
+ * val a = createOnCreateEvent(null)
+ * val b = createOnCreateEvent(Bundle())
+ * assertThat(a <= b).isTrue()
+ * assertThat(b <= a).isTrue()
+ * ```
+ *
+ * The equality does not hold:
+ * ```
+ * assertThat(a == b).isFalse()
```

Comment:
Should they be?

Reply:
IMO, no, because Android bundles also do not implement equal